es6/docxtemplater.js

```diff
@@ -28,15 +28,26 @@ const {
 const currentModuleApiVersion = [3, 21, 0];
 
 const Docxtemplater = class Docxtemplater {
-	constructor() {
-		if (arguments.length > 0) {
+	constructor(zip, { modules = [] } = {}) {
```

Comment:
I just found out that there is something quite important missing with this new API.

docxtemplater allows to setOptions, for example change the delimiters with following code : 

```
var doc = new Docxtemplater();
doc.loadZip(zip);
doc.setOptions({delimiters: {start: '[[', end: ']]'}});
doc.compile();
// So that you can write : [[user.name]] instead of {user.name} in your template
```

However, with the new API, this is currently not possible, because the compilation is done during construction : 

If you write : 

```
var doc = new Docxtemplater(zip);
doc.setOptions({delimiters: {start: '[[', end: ']]}});
```

your template was already compiled at the time when you call setOptions.

So the idea would be to be able to pass in options as part of the second argument, like this : 

```
var doc = new Docxtemplater(zip, {delimiters: {start:'[[', end:']]'}});
```

or with some modules also : 

```
var doc = new Docxtemplater(zip, {delimiters: {start:'[[', end:']]'}, modules: [ new HTMLModule() ] });
```

Do you want to do that change in this Pull Request ? I can also do it on my side before releasing 3.17.0. This implies changes in the changelog/documentation/tests.

Reply:
For the test, you could copy one test from there : https://github.com/open-xml-templating/docxtemplater/blob/master/es6/tests/base.js#L646

Reply:
Although I would ideally like to do it in another PR, that would mean us pushing the bug with this PR. So, It would be right to do it in this PR only.

I would also like to suggest the change in the way we can pass options:

```
const options = {delimiters: {start:'[[', end:']]'}}
const modules = [ new HTMLModules() ];
const doc = new Docxtemplater(zip, {options, modules})
```
This was even if we decide to add more customization in options, we would not have to change constructor. What are your thoughts on that?

Reply:
Instead of `options`, we can call it `configurations` because the outer object is being called `options`?

Reply:
> This was even if we decide to add more customization in options, we would not have to change constructor. What are your thoughts on that?

modules can be regarded as an option, so we could write our code like this : 

```
constructor(zip, options = {}) {
    this.compiled = {};
    options.modules = options.modules || [];
    this.modules = [commonModule()];
    this.setOptions(options);
    options.modules.forEach(module => {
        this.attachModule(module);
    });
    if (zip) {
        if (!zip.files || typeof zip.file !== "function") {
            throw new Error(
                "The first argument of docxtemplater's constructor must be a valid zip file (jszip v2 or pizzip v3)"
            );
        }
        this.loadZip(zip).compile();
    }
```

Reply:
modules can be part of the options, I find it simpler than to create a new level "customizations"

Reply:
Okay, then I will remove the `modules` from `options` before calling `setOptions`, because It will be unused anyway.

Reply:
Yes, that works for me.

Reply:
I suggest to use destructuring in the constructor to do this : 

```
constructor(zip, { modules = [], ...options} = {})
```

Reply:
Yes, a good idea.

Reply:
Will the `setOptions` be removed in the v4 version?

Reply:
The `setOptions` is not available in `api.rst`. I am planning to add it to the documentation also.

Reply:
Yes setOptions will also be removed in version v4.

Ok for the doc on setOptions. You can link to the specific page for options for the doc about this method : https://docxtemplater.readthedocs.io/en/latest/configuration.html

Reply:
Completed it.